scripts/migrations/2020-10-archivematica_bags_migration.py

```diff
@@ -0,0 +1,246 @@
+            new_checksums = _generate_updated_checksums()
+
+            merged_checksums = dict(existing_checksums.items() | new_checksums.items())
+
+            with open(f"{working_folder}/{tagmanifest_name}", "w") as fp:
+                for checksum, filename in merged_checksums.items():
+                    fp.write(f"{filename} {checksum}\n")
+
+        def _get_archivematica_uuid():
+            files = _filter_bag_files("data/METS.")
+
+            assert len(files) == 1
+            mets_file_with_id = files[0]
+
+            archivematica_uuid = mets_file_with_id.split("/METS.")[-1].split(".xml")[0]
+
+            assert archivematica_uuid
+            return archivematica_uuid
+
+        def _update_bag_info():
+            archivematica_uuid = _get_archivematica_uuid()
+            with open(f"{working_folder}/bag-info.txt", "a") as fp:
+                fp.write(f"Internal-Sender-Identifier: {archivematica_uuid}")
+
+        def _upload_bag_to_s3():
+            workflow_s3_client.upload_file(
+                f"{target_folder}/{archive_name}",
+                s3_upload_bucket,
+                f"{s3_upload_prefix}/{archive_name}",
+            )
+
+        def _request_ingest():
+            response = storage_client.create_s3_ingest(
+                space=space,
+                external_identifier=external_identifier,
+                s3_bucket=s3_upload_bucket,
+                s3_key=f"{s3_upload_prefix}/{archive_name}",
+                ingest_type="update",
+            )
+
+        _write_fetch_file()
+        _get_bag_files()
+        _update_bag_info()
+        _write_new_tagmanifest()
+        _compress_bag()
+        _upload_bag_to_s3()
+        _request_ingest()
+
+        sys.exit(1)
+
+    return create_bag
+
+
+if __name__ == "__main__":
+    storage_role_arn = "arn:aws:iam::975596993436:role/storage-developer"
+    workflow_role_arn = "arn:aws:iam::299497370133:role/workflow-developer"
+
+    elastic_secret_id = "archivematica_bags_migration/credentials"
+    index = "storage_bags"
+
+    environment_id = "stage"
+
+    environments = {
+        "prod": {
+            "bucket": "wellcomecollection-archivematica-ingests",
+            "api_url": "https://api.wellcomecollection.org/storage/v1",
+        },
+        "stage": {
+            "bucket": "wellcomecollection-archivematica-staging-ingests",
+            "api_url": "https://api-stage.wellcomecollection.org/storage/v1",
+        },
+    }
+
+    api_url = environments[environment_id]["api_url"]
+
+    workflow_s3_client = get_aws_client(resource="s3", role_arn=workflow_role_arn)
+
+    storage_s3_client = get_aws_client(resource="s3", role_arn=storage_role_arn)
+
+    elastic_client = get_elastic_client(
+        role_arn=storage_role_arn, elastic_secret_id=elastic_secret_id
+    )
+
+    storage_client = get_storage_client(api_url=api_url)
+
+    # TODO: update query to include born-digital-accessions
```

Comment:
And to ignore bags that already have an Internal-Sender-Identifier field, so you don't send multiple updates for the same bag.